### ISSUE TITLE
Memory leak in firefox Pc

### ISSUE THREAD

**Issue by pityalvarez** [Arcus92/instagram-video-control#40]
On a profile, after I open a video, and scroll through the profile I notice how laggy it is. I check task manager 2gb-3gb when it's a couple of instagram pages.
The second i deactivate the add-on/extension everything works as normal.

**Comment by raresvaduva** [comment]
Yep, I've also been having this recently, I look at a lot of stuff on insta, like news and stuff, and suddenly I notice the browser just starts completely freezing up, can't switch tabs, can't close tabs, it just doesn't react to anything, or really really laggy, like at 0.001 speed.

Then, when you do try to close it, it also doesn't react and just hangs there, after some time though, the window closes on it's own, but the processes are still active in task manager. After a further 3-4 minutes, all firefox processes also finally close down, and usually the firefox crash reporter comes up.


Steps to reproduce (kinda):
1. Open up instagram.
2. Go open a bunch of video reels (10-20)
3. Load the tabs, actively switch between them, unmute them, go around pressing pause/play, seek further, closer etc.
4. At some point, the video player stops, or it only works to a certain time (like on a 2 min video, until sec 42 it works, but at second 42 it just stops completely), then after that the browser starts freezing up AND, if you have task manager open, you can see memory usage creeping up quite fast.

It seems (maybe) that longer video reels contribute more to this? Try this with more 1:30, 2+ mins video reels...

Would really really really really wish this would get fixed :( .
Thanks!
